### ISSUE TITLE
FEAT: Add option to change default from "Run All" to "Run Current"

### ISSUE THREAD

**Issue by h** [beekeeper-studio/beekeeper-studio#1679]
**Is your feature request related to a problem? Please describe.**
Our employees sometimes run unintended database updates in Beekeeper by accident. That's because people often have multiple queries in the same editor tab, and by default Beekeeper runs ALL queries in the current editor, rather than just the query under the cursor. This makes it far too easy to accidentally run a previous UPDATE or DELETE query that was written in a different section of the SQL editor.

**Describe the solution you'd like**
Please consider swapping the `Run` vs `Run Current` buttons. `Run Current` is the "safe" option and therefore should be the one that's easy to access. The `Run` button should be hidden behind the dropdown, which is where the `Run Current` button is presently located.

<img width="269" alt="Screenshot 2023-08-22 at 3 23 07 PM" src="https://github.com/beekeeper-studio/beekeeper-studio/assets/3854874/d88c479a-3988-410f-8a7b-d25aa74867b6">

By the same token, the `⌘Enter→Run and `⇧⌘Enter→Run Current` keyboard shortcuts should be swapped. Rationale: the "safe" option (i.e. "Run Current") should be the one that's more easily accessed.

**Describe alternatives you've considered**
If you're concerned about backward compatibility, that can be addressed by adding a setting to enable the old behavior.

**Additional context**
We've had occurrences of data loss because somebody accidentally ran ALL queries in the current editor, rather than just the current one. I've now resorted to telling our engineers to use Postico 2 which doesn't have this problem. As an Open Source developer myself, I'd much prefer using an open source solution like BeeKeeper. We've just been bitten by this issue too many times.

Most other SQL editors (e.g. Postico 2, etc) use the keyboard shortcut & button suggested here, so implementing this feature request would bring Beekeeper in line with the behavior most developers are already used to.

**Comment by rizqikazukun** [comment]
I am agree, since i migrating from other tools, it's hard to not executing delete query that may harmful if use at production level.

**Comment by PHLAK** [comment]
I thought I had opened an issue for this before that got closed as "wont fix", however I cannot find it now if in fact I did. But I agree with this, rarely do I want to run _all_ queries I have in the editor window at once.

**Comment by gsigler** [comment]
I completely agree with this. Its far too easy to run your whole file.

**Comment by rizqikazukun** [comment]
maybe it's hard for existing user, but i suggest to make a preferences menu that there a preference for keybinding, i can look up the source code keybinding, but i can't write the code 😅 i don't have the basic

**Comment by PHLAK** [comment]
The ability to customize the key bind would be perfect.

**Comment by not-night-but** [comment]
Yeah I think the solution to this is adding a config option. We have configuration implemented and getting ready to merge, so that can be added fairly easily

**Comment by honestlysamuk** [comment]
+1 on this.

**Comment by gsigler** [comment]
++1, why make it harder to do the safe thing of running one query. Make it harder to run the whole file

**Comment by not-night-but** [comment]
@gsigler we're working on a fix for this. I'm personally hesitant to change defaults for the application, especially when it comes to what queries are being run. I believe right now it is possible to swap the keybinds yourself, but we're also looking at an option to possibly switch the default on the button. How would we feel about that?

**Comment by PHLAK** [comment]
Though I still think the default should be the safer option, I wouldn't really care as long as we had the option to change/customize the binding. That would satisfy me.

**Comment by gsigler** [comment]
> [@gsigler](https://github.com/gsigler) we're working on a fix for this. I'm personally hesitant to change defaults for the application, especially when it comes to what queries are being run. I believe right now it is possible to swap the keybinds yourself, but we're also looking at an option to possibly switch the default on the button. How would we feel about that?

I think that would be great. Making it customizable makes it so it can work well for you no matter what 'muscle' memory you have.

**Comment by not-night-but** [comment]
@gsigler So you can change the keybinds for it right now (although it looks like we forgot to update the keybind hint so that won't change when you do this, I will update that).

In your [config](https://docs.beekeeperstudio.io/user_guide/configuration/), add the following:

```ini
[keybindings.queryEditor]
submitTabQuery = ctrlOrCmd+shift+enter
submitCurrentQuery = ctrlOrCmd+enter
```

This will override the default keybinds, so `f5` and `shift+f5` will no longer work, but you can add those to the config as well if you want:

```ini
[keybindings.queryEditor]
submitTabQuery[] = ctrlOrCmd+shift+enter
submitTabQuery[] = f5
submitCurrentQuery[] = ctrlOrCmd+enter
submitCurrentQuery[] = shift+f5
```

**Comment by rathboma** [comment]
Ok, I'm making a decision on this for 6.7:

1. Will make the default run current query
2. Will make it easier to change the 'primary' action in the configuration (not just swap keymaps)

I think y'all are right. Run current is the safer default, might confuse people to start with, but it's less dangerous and goes along with other app conventions

**Comment by honestlysamuk** [comment]
6 7
